README.rst

```diff
@@ -224,6 +224,66 @@ is no natural place for the annotation to live. Here is one possibility:
     class Leaf(Node): ...
     class Branch(Node): ...
 
+Just use Union
```

Comment:
I think that the title of this section could be a little more formal declarative, rather than phrased as a command to the read.

Maybe something like

- `Union` of variants
- `Union` of subtypes

Reply:
Changed it to "``Union`` of independent variants". I am not super happy with the heading. I'll keep thinking about it.